plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsBlobStore.java

```diff
@@ -118,7 +119,8 @@ private Path translateToHdfsPath(BlobPath blobPath) {
         try {
             return AccessController.doPrivileged((PrivilegedExceptionAction<V>)
                     () -> operation.run(fileContext), null, new ReflectPermission("suppressAccessChecks"),
-                     new AuthPermission("modifyPrivateCredentials"), new SocketPermission("*", "connect"));
+                     new AuthPermission("modifyPrivateCredentials"), new SocketPermission("*", "connect"),
+                     new AuthPermission("doAs"), new ServicePermission("*", "initiate,accept"));
```

Comment:
I think that the comment on this method is no longer accurate. Can you adjust it?